pilot/pkg/config/kube/crd/config.go

```diff
@@ -33,37 +31,11 @@ func (in *IstioKind) GetSpec() map[string]interface{} {
 	return in.Spec
 }
 
-// SetSpec for a wrapper
-func (in *IstioKind) SetSpec(spec map[string]interface{}) {
-	in.Spec = spec
-}
-
 // GetObjectMeta from a wrapper
 func (in *IstioKind) GetObjectMeta() meta_v1.ObjectMeta {
 	return in.ObjectMeta
 }
 
-// SetObjectMeta for a wrapper
-func (in *IstioKind) SetObjectMeta(metadata meta_v1.ObjectMeta) {
-	in.ObjectMeta = metadata
-}
-
-// IstioKindList is the generic Kubernetes API list wrapper
-type IstioKindList struct {
-	meta_v1.TypeMeta `json:",inline"`
-	meta_v1.ListMeta `json:"metadata"`
-	Items            []IstioKind `json:"items"`
-}
-
-// GetItems from a wrapper
-func (in *IstioKindList) GetItems() []IstioObject {
-	out := make([]IstioObject, len(in.Items))
-	for i := range in.Items {
-		out[i] = &in.Items[i]
-	}
-	return out
-}
-
 // DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
 func (in *IstioKind) DeepCopyInto(out *IstioKind) {
```

Comment:
We may want to remove the use of IstioKind and use kubetypes in client go.

Reply:
Yeah I want to clean this up but there is pretty extensive code using this. I can do it in a follow-up - this pr is huge